validator/sawtooth_validator/networking/dispatch.py

```diff
@@ -160,15 +175,90 @@ def add_handler(self, message_type, handler, executor, priority=None):
         if priority is not None:
             self._priority[message_type] = priority
 
+    def set_preprocessor(self, message_type, preprocessor, executor):
```

Comment:
The type expected for preprocessor should be clearly documented here in a docstring. I would expect it to be something like a function `fn(message_content: bytes) -> PreprocessorResult` where `PreprocessorResult` informs the dispatcher what to do next and wraps the processed message content.